Sources/SwiftParser/IncrementalParseTransition.swift

```diff
@@ -43,36 +43,15 @@ extension Parser {
 ///
 /// This is also used for testing purposes to ensure incremental reparsing
 /// worked as expected.
-public protocol IncrementalParseReusedNodeDelegate {
-  /// Accepts ``Syntax`` node of skipped source region.
-  ///
-  /// - Parameters:
-  ///   - previousNode: The node from the previous tree that is associated with
-  ///                   the skipped source region.
-  func parserReusedNode(previousNode: Syntax)
-}
-
-/// An implementation of `IncrementalParseReusedNodeDelegate` that just collects
-/// the range and re-used node into an array.
-public final class IncrementalParseReusedNodeCollector:
-  IncrementalParseReusedNodeDelegate
-{
-  public var nodes: [Syntax] = []
-
-  public init() {}
-
-  public func parserReusedNode(previousNode: Syntax) {
-    nodes.append(previousNode)
-  }
-}
+public typealias ReusedNodeCallback = (_ nodes: Syntax) -> ()
```

Comment:
The callback only takes a single node.

```suggestion
public typealias ReusedNodeCallback = (_ node: Syntax) -> ()
```